optional/cloudfront_cdn/main.tf

```
@@ -55,6 +63,83 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
   }
 
   viewer_certificate {
-    cloudfront_default_certificate = true
+    # This part is very ugly because there are 3 typical use cases:
+    #   1. Using the default certificate domain.
+    #      In this case, no additional resources are needed because the CloudFront default certificate is sufficient.
+    #   2. Use the existing ACM certificate from the `us-east-1` region.
+    #      In this case, only the domain record is created.
+    #   3. Generate a new ACM certificate in the `us-east-1` region.
+    #      Applicable when the main certificate already exists in another region.
+    cloudfront_default_certificate = var.aliases == null ? true : false
+
+    acm_certificate_arn = (
+                            var.aliases == null
+                              ? null
+                              : var.alias_certificate_arn == null
+                                ? aws_acm_certificate.alias_certificate[0].arn
+                                  : var.alias_certificate_arn
```

I suggest this so the configuration is consistent with the above comments:
```suggestion
                              : var.alias_certificate_arn != null
                                ? var.alias_certificate_arn
                                  : aws_acm_certificate.alias_certificate[0].arn
```